library/x509.c

```diff
@@ -775,17 +775,24 @@ int mbedtls_x509_dn_gets( char *buf, size_t size, const mbedtls_x509_name *dn )
             ret = mbedtls_snprintf( p, n, "\?\?=" );
         MBEDTLS_X509_SAFE_SNPRINTF;
 
-        for( i = 0; i < name->val.len; i++ )
+        for( i = 0, j = 0; i < name->val.len; i++, j++ )
         {
-            if( i >= sizeof( s ) - 1 )
+            if( j >= sizeof( s ) - 1 )
```

Comment:
I'm a bit concerned that this will silently truncate the output if it doesn't fit - should we return an error in this case instead? (Yes, I know this is pre-existing behaviour, but still, it seems pretty horrible...)